src/components/editor/Store/create.ts

```diff
@@ -105,18 +124,58 @@ const getInitialState = <T>(
             );
         },
 
-        setStatus: (newVal) => {
+        setStatus: (value, path) => {
             set(
                 produce((state) => {
-                    state.isSaving = newVal === EditorStatus.SAVING;
-                    state.isEditing = newVal === EditorStatus.EDITING;
-                    state.status = newVal;
+                    if (Object.hasOwn(state.statuses, path)) {
+                        state.statuses[path] = value;
+                    } else {
+                        state.statuses = { ...state.statuses, [path]: value };
+                    }
+
+                    const editorStatuses: EditorStatus[] = Object.values(
+                        state.statuses
+                    );
+
+                    state.isSaving = editorStatuses.some(
+                        (status) => status === EditorStatus.SAVING
+                    );
+                    state.isEditing = editorStatuses.some(
+                        (status) => status === EditorStatus.EDITING
+                    );
+
+                    state.invalidEditors = Object.entries(state.statuses)
+                        .filter(
+                            ([_key, status]) =>
+                                status === EditorStatus.INVALID ||
+                                status === EditorStatus.SAVE_FAILED ||
+                                status === EditorStatus.OUT_OF_SYNC
+                        )
+                        .map(([key, _status]) => key);
                 }),
                 false,
                 'Setting status'
             );
         },
 
+        removeStaleStatus: (value) => {
+            set(
+                produce((state) => {
+                    const { invalidEditors, statuses } = get();
+
+                    if (Object.hasOwn(statuses, value)) {
+                        state.statuses = omit(statuses, value);
+                    }
+
+                    state.invalidEditors = invalidEditors.filter(
+                        (path) => path !== value
+                    );
+                }),
+                false,
+                'Stale Status Removed'
+            );
+        },
```

Comment:
I feel like this function should not exist _in this way_. I feel like this update should be auto handled whenever `removeAttribute` is called.

Reply:
I feel the same way. This was something I would have discussed in dailies this past Monday. _Checkpoint comment. Need to pivot to something else..._